### PR TITLE
Port to meson build system #2

### PR DESCRIPTION
**Pull request by t-chaik**
https://github.com/hughsie/PackageKit/pull/309 follow-up.

### CODE REVIEW

**Comment by hughsie** [comment]
Wow, do you want to squash all these down and then I can do some proper review? Thanks.

**Comment by t-chaik** [comment]
> Wow, do you want to squash all these down and then I can do some proper review? Thanks.

Do you want me to squash everything in let say 3 commits (meson port, autotools removal and CI update) or just my new ones in jurst one?

**Comment by t-chaik** [comment]
Tests from master (autotools) are failing on my machine too (`make check`). Correct me if I'm wrong, but it seems like current CI is not running them but only perform build + install. For what I can tell, they are not failing for environment/build reasons and fixing them would require jumping in PackageKit's code and understand that's going wrong. I've drop them from the meson CI for now, please advice on what you'd prefer here.

**Comment by hughsie** [comment]
I think disable the failing self tests for now, I think the meson patch is more important. When it's all passing CI I can fix up the tests as required. Thanks again!

File: docs/api/dbus/meson.build [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+packagekit_ref_xml = custom_target(
+	'org.freedesktop.PackageKit.ref.xml',
+	input: join_paths(meson.source_root(), 'src', 'org.freedesktop.PackageKit.xml'),
+	output: 'org.freedesktop.PackageKit.ref.xml',
+	command: [
+		spec_to_docbook,
+		files('spec-to-docbook.xsl'),
+		'@INPUT@',
+		'@OUTPUT@',
+	],
```
Comment (crawfxrd) [inline docs/api/dbus/meson.build]: Adding `build_by_default: true` to these custom targets will allow `ninja dist` to work.

**Comment by t-chaik** [comment]
Shall I squash all this in 2 or 3 commits (+meson, -autotools, +CI)?

**Comment by hughsie** [comment]
Ya, squash please.

**Comment by hughsie** [comment]
I'm just fixing up some of the failing tests this morning.

**Comment by hughsie** [comment]
I've fixed up a lot of the issues, so lets continue in https://github.com/hughsie/PackageKit/pull/365 -- all comments there very welcome, thanks!

**Pull request closed without merging**